goldens/by_regex.in

```diff
@@ -0,0 +1,40 @@
+No capturing group
+  keep-sorted-test start by_regex=int|bool|long
+  int baz
+  bool foo
+  long bar
+  keep-sorted-test end
+
+Capturing group
+  keep-sorted-test start by_regex=['[^ ]+ (.*)']
+  int baz
+  bool foo
+  long bar
+  keep-sorted-test end
+
+Capturing group and non-capturing group
+  keep-sorted-test start by_regex=['(?:int|bool|long) (.*)']
+  int baz
+  bool foo
+  long bar
+  keep-sorted-test end
+
+Multiple regexes
+  keep-sorted-test start by_regex=['(?:int|bool|long) (.*)', 'int|bool|long']
+  int baz
+  long baz
+  bool baz
+  bool foo
+  long foo
+  int foo
+  long bar
+  bool bar
+  int bar
+  keep-sorted-test end
+
+Cannot combine with ignore_prefixes
+  keep-sorted-test start by_regex=.* ignore_prefixes=foo,bar
+  2
+  1
+  3
+  keep-sorted-test end
```

Comment:
Shall we add some edge cases, for example, a regex can contain `,`, `'`, `[` or `]`.